### PR TITLE
Pricing box test

### PR DESCRIPTION
**Pull request by acostacos**
Tests do not pass in the frontend due to the following issue: https://github.com/gambitph/Stackable/issues/1069

### CODE REVIEW

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 1]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+			'height': '300px',
+		},
+	} )
+	// TO DO: Make assert align for image
+
+	// Title, Price, Price Prefix, Price Suffix, Sub Price and Description Tabs
+	const typographyAssertions = [ 'title', 'price', 'price-prefix', 'price-suffix', 'subprice', 'description' ]
+	typographyAssertions.forEach( typographyAssertion => {
+		// Handle Sub Price label
+		const label = typographyAssertion === 'subprice' ? 'Sub Price' : typographyAssertion.split( '-' ).map( word => startCase( word ) ).join( ' ' )
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:174]: `startCase` function can already handle kebab casing. No need to split the words

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 2]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+		cy.collapse( label )
+
+		if ( typographyAssertion === 'title' ) {
+			desktopOnly( () => {
+				cy.adjust( 'Title HTML Tag', 'h4' ).assertHtmlTag( '.ugb-pricing-box__title', 'h4' )
+			} )
+		}
+
+		desktopOnly( () => {
+			const colorTitle = ( ! typographyAssertion.includes( 'price' ) ) ? label : 'Text'
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:185]: `includes` function is often used in searching array values. Let's use `.match(/price/)` instead

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 3]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+		} )
+		cy.adjust( 'Size', 2, { viewport, unit: 'em' } ).assertComputedStyle( {
+
+			[ `.ugb-pricing-box__${ textClass }` ]: {
+				'font-size': '2em',
+			},
+		} )
+
+		// Only for Title, Price, Sub Price and Description
+		if ( ! typographyAssertion.includes( 'price-' ) ) {
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:207]: Use `match` function here also

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 4]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+		} )
+
+		const textClass = typographyAssertion === 'price' ? 'price-line' : typographyAssertion
+		cy.adjust( 'Size', 55, { viewport, unit: 'px' } ).assertComputedStyle( {
+			[ `.ugb-pricing-box__${ textClass }` ]: {
+				'font-size': '55px',
+			},
+		} )
+		cy.adjust( 'Size', 2, { viewport, unit: 'em' } ).assertComputedStyle( {
+
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:200]: remove unnecessary newline

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 5]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+			[ `.ugb-pricing-box__${ textClass }` ]: {
+				'font-size': '2em',
+			},
+		} )
+
+		// Only for Title, Price, Sub Price and Description
+		if ( ! typographyAssertion.includes( 'price-' ) ) {
+			assertTypography( `.ugb-pricing-box__${ textClass }`, { viewport } )
+		} else {
+			desktopOnly( () => {
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:210]: I've seen a lot of typography options that have fewer options inside. Maybe we can pass more options in `assertTypography` to disable some options?

Example code:
```
assertTypography( `.ugb-pricing-box__${ textClass }, { viewport, disableLetterSpacing: true  }` )
```

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 6]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+				'Line-Height': { value: 2, unit: 'em' },
+			}, { viewport } ).assertComputedStyle( {
+				[ `.ugb-pricing-box__${ textClass }` ]: {
+					'font-size': '3em',
+					'line-height': '2em',
+				},
+			} )
+		}
+		// Only for Title, Sub Price, and Description
+		if ( ! typographyAssertion.includes( 'price' ) || typographyAssertion === 'subprice' ) {
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:241]: This statement might be confusing. Instead we can use this:
```
if ( Array( 'title', 'subprice', 'description' ).includes( typographyAssertion ) )
```

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 7]
```diff
@@ -53,3 +59,317 @@ function switchDesign() {
...
+		},
+	} )
+
+	assertAligns( 'Align', '.ugb-button-container', { viewport } )
+
+	// Effects Tab
+	cy.collapse( 'Effects' )
+
+	desktopOnly( () => {
+		cy.adjust( 'Hover Effect', 'lift-more' ).assertClassName( '.ugb-pricing-box__item', 'ugb--hover-lift-more' )
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:367]: assert all hover effects here

**Review by blankart — CHANGES_REQUESTED** [review]

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 8]
```diff
@@ -59,6 +63,311 @@ function switchDesign() {
...
+		cy.adjust( 'Size', 2, { viewport, unit: 'em' } ).assertComputedStyle( {
+			[ `.ugb-pricing-box__${ textClass }` ]: {
+				'font-size': '2em',
+			},
+		} )
+
+		let typographySettings = {}
+		// Only for Price Prefix and Price Suffix
+		if ( typographyAssertion.match( '/price-/' ) ) {
+			typographySettings = {
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:212]: make the `typographySettings` constant

directly mutate the properties instead e.g.
```
object.property = 'changed value'
```

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 9]
```diff
@@ -59,6 +63,311 @@ function switchDesign() {
...
+		cy.adjust( 'Text Color', '#ffa03b' )
+		cy.adjust( 'Hover Effect', 'scale' ).assertClassName( '.ugb-button', 'ugb--hover-effect-scale' )
+		cy.adjust( 'Hover Opacity', 0.6 )
+		cy.adjust( 'Hover Colors', {
+			'Button Color #1': '#bd8b8b',
+			'Button Color #2': '#3fa35b',
+			'Gradient Direction (degrees)': 72,
+			'Text Color': '#80194d',
+		} )
+		cy.adjust( 'Typography', {
```
Comment (blankart) [inline cypress/integration/v2/blocks/pricing-box.spec.js:272]: use `assertTypography` here as well

**Comment by blankart** [comment]
Approved.

Pull master after merging #101. Ask @legaspielaine to run the tests.

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 10]
```diff
@@ -59,6 +63,299 @@ function switchDesign() {
...
+		} )
+		cy.adjust( 'Flip Shape Horizontally', true )
+		cy.adjust( 'Flip Shape Vertically', true )
+		cy.adjust( 'Stretch Shape Mask', true ).assertClassName( 'img.ugb-img--shape', 'ugb-image--shape-stretch' )
+
+		// We won't be able to assert image size for now since it requires server handling.
+	} )
+
+	cy.adjust( 'Image Width', 300, { viewport } )
+	cy.adjust( 'Force square image', true, { viewport } ).assertComputedStyle( {
```
Comment (legaspielaine) [inline cypress/integration/v2/blocks/pricing-box.spec.js:166]: Force square image does not have responsive options - put this inside desktopOnly

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 11]
```diff
@@ -59,6 +63,299 @@ function switchDesign() {
...
+		} ).assertClassName( '.ugb-button', 'ugb-button--icon-position-right' )
+	} )
+
+	assertTypography( '.ugb-button .ugb-button--inner', { viewport }, {
+		enableLineHeight: false,
+	} )
+	assertAligns( 'Align', '.ugb-button-container', { viewport } )
+
+	// Effects Tab
+	cy.collapse( 'Effects' )
```
Comment (legaspielaine) [inline cypress/integration/v2/blocks/pricing-box.spec.js:331]: we can add this inside desktopOnly

File: cypress/integration/v2/blocks/pricing-box.spec.js [hunk 12]
```diff
@@ -59,6 +63,299 @@ function switchDesign() {
...
 }
 
+function styleTab( viewport, desktopOnly, registerBlockSnapshots ) {
+	cy.setupWP()
+	cy.newPage()
+	cy.addBlock( 'ugb/pricing-box' ).as( 'pricingBoxBlock' )
+	const pricingBoxBlock = registerBlockSnapshots( 'pricingBoxBlock' )
+	cy.openInspector( 'ugb/pricing-box', 'Style' )
+
+	//General Tab
```
Comment (legaspielaine) [inline cypress/integration/v2/blocks/pricing-box.spec.js:73]: add space after //

**Review by blankart — APPROVED** [review]

**Pull request merged**